pkg/reconciler/kubernetes/tektonhub/tektonhub.go

```diff
@@ -0,0 +1,595 @@
+			Labels: map[string]string{
+				"app": "api",
+			},
+		},
+		Data: map[string]string{
+			"CONFIG_FILE_URL": th.Spec.Api.HubConfigUrl,
+		},
+	}
+}
+
+func createDbSecret(name, namespace string, existingSecret *corev1.Secret) *corev1.Secret {
+	s := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			Labels: map[string]string{
+				"app": "db",
+			},
+		},
+		Type: corev1.SecretTypeOpaque,
+	}
+
+	if existingSecret != nil && existingSecret.Data != nil {
+		s.Data = existingSecret.Data
+	}
+
+	s.StringData = make(map[string]string)
+
+	if s.Data["POSTGRES_DB"] == nil || len(s.Data["POSTGRES_DB"]) == 0 {
+		s.StringData["POSTGRES_DB"] = "hub"
+	}
+
+	if s.Data["POSTGRES_USER"] == nil || len(s.Data["POSTGRES_USER"]) == 0 {
+		s.StringData["POSTGRES_USER"] = "hub"
+	}
+
+	if s.Data["POSTGRES_PASSWORD"] == nil || len(s.Data["POSTGRES_PASSWORD"]) == 0 {
+		s.StringData["POSTGRES_PASSWORD"] = "hub"
+	}
+
+	if s.Data["POSTGRES_PORT"] == nil || len(s.Data["POSTGRES_PORT"]) == 0 {
+		s.StringData["POSTGRES_PORT"] = "5432"
+	}
+
+	return s
+}
+
+func (r *Reconciler) applyManifest(ctx context.Context, manifestLocation string, th *v1alpha1.TektonHub, installerSetName, version, prefixName string) error {
+	manifest := r.manifest.Append()
+	logger := logging.FromContext(ctx)
+
+	if err := common.AppendManifest(&manifest, manifestLocation); err != nil {
+		return err
+	}
+
+	manifest = manifest.Filter(mf.Not(mf.Any(mf.ByKind("Secret"), mf.ByKind("Namespace"), mf.ByKind("ConfigMap"))))
+
+	manifest, err := manifest.Transform(
+		mf.InjectOwner(th),
+		mf.InjectNamespace(namespace),
+	)
+	if err != nil {
+		logger.Error("failed to transform manifest")
+		return err
+	}
+
+	if err := createInstallerSet(ctx, r.operatorClientSet, th, manifest,
+		version, installerSetName, prefixName, namespace); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *Reconciler) installed(ctx context.Context, instance v1alpha1.TektonComponent) (*mf.Manifest, error) {
+	// Create new, empty manifest with valid client and logger
+	installed := r.manifest.Append()
+	stages := common.Stages{common.AppendInstalled}
+	err := stages.Execute(ctx, &installed, instance)
+	return &installed, err
+}
+
+// checkIfInstallerSetExist checks if installer set exists for a component and return true/false based on it
+// and if installer set which already exist is of older version then it deletes and return false to create a new
+// installer set
+func checkIfInstallerSetExist(ctx context.Context, oc clientset.Interface, relVersion string,
+	th *v1alpha1.TektonHub, component string) (bool, error) {
+
+	// Check if installer set is already created
+	compInstallerSet, ok := th.Status.HubInstallerSet[component]
+	if !ok {
+		return false, nil
+	}
+
+	if compInstallerSet != "" {
+		// if already created then check which version it is
+		ctIs, err := oc.OperatorV1alpha1().TektonInstallerSets().
+			Get(ctx, compInstallerSet, metav1.GetOptions{})
```

Comment:
We need to use List() instead of Get to verify this. ie, use a set of labels to query for Installersets instead of using a constant name.

If more than one InstallerSet is returned for a type delete all and create a new one.

As installersets are being create using `GenerateName: fmt.Sprintf("%s-", prefix),` using names and Get queries is not enough to catch duplicates and obsolete installersets. 

eg: 
- https://github.com/tektoncd/operator/blob/a0cd67e5c428babf1f76316a14cf9c4526055cab/pkg/reconciler/kubernetes/tektonpipeline/tektonpipeline.go#L160
- 
https://github.com/tektoncd/operator/blob/a0cd67e5c428babf1f76316a14cf9c4526055cab/pkg/reconciler/kubernetes/tektonpipeline/tektonpipeline.go#L359